pkg/controllers/nodeclaim/disruption/drift.go

```diff
@@ -81,6 +84,21 @@ func (d *Drift) isDrifted(ctx context.Context, nodePool *v1.NodePool, nodeClaim
 	}); reason != "" {
 		return reason, nil
 	}
+	// Mark the NodeClaim as drifted due to a stale instance type if
```

Comment:
Can we define what we mean by a state instance type? It would be nice to have the definition as a comment

Reply:
Does the 3 definitions below help? It's mostly "does the instance type with the offering exist anymore in the instance type offerings cache", if not it's stale